src/averageAges.js

```diff
@@ -55,7 +75,24 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const sonCondition = (child) => child.sex === 'm';
+  const relativesCondition = (mothersArr, child) => mothersArr
+    .some(mother => mother.name === child.mother);
+
+  const mothers = people
+    .filter(person => people
+      .some(child => child.mother === person.name));
+  const children = people
+    .filter(child => onlyWithSon ? sonCondition(child)
+  && relativesCondition(mothers, child)
+      : relativesCondition(mothers, child));
+
+  const ages = children
+    .map(child => child.born - mothers
+      .find(mother => mother.name === child.mother).born);
```

Comment:
You can already use **reduce** method for to reduce the number of methods called:
```suggestion
  const ages = children
    .reduce((a, b)=> a += b.born - mothers
      .find(mother => mother.name === b.mother).born, 0);
```